### PR TITLE
[fix bug 1295693] Add headless http status integration tests for pages that rely on external data

### PR DESCRIPTION
**Pull request by alexgibson**
@pmac @jgmize r?

I ended up deciding to test only the main pages that list generated content, as opposed to exhaustively parsing the links in each page as it ended up generating thousands of tests. I think this should generally give us enough confidence to alert us when something is broken, although I'd be happy to expand it a bit further if we think it's worth it.

### CODE REVIEW

File: tests/functional/test_generated_content_pages.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        '/security/known-vulnerabilities/thunderbird-3.0/',
+        '/security/known-vulnerabilities/thunderbird-2.0/',
+        '/security/known-vulnerabilities/thunderbird-1.5/',
+        '/security/known-vulnerabilities/thunderbird-1.0/',
+        '/security/known-vulnerabilities/seamonkey-2.0/',
+        '/security/known-vulnerabilities/seamonkey-1.1/',
+        '/security/known-vulnerabilities/seamonkey-1.0/',
+        '/security/known-vulnerabilities/mozilla-suite/',
+        '/security/known-vulnerabilities/older-vulnerabilities/'
+        )
```
Comment (pmac) [inline tests/functional/test_generated_content_pages.py]: Super NIT: Deindent one level.

File: tests/functional/test_generated_content_pages.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    argvalues = []
+    for path in paths:
```
Comment (pmac) [inline tests/functional/test_generated_content_pages.py]: Could be done as a list comprehension:

```python
metafunc.parametrize('url', [base_url + path for path in paths])
```

**Comment by pmac** [comment]
LGTM! Thanks Alex! I agree that this list should give us a good deal more confidence.

**Comment by pmac** [comment]
Just a couple nits. You can change them or not. r+

File: tests/functional/test_generated_content_pages.py (hunk 2 above)
Comment (alexgibson) [inline tests/functional/test_generated_content_pages.py]: Nice 👍

**Comment by alexgibson** [comment]
Thanks @pmac updated

**Comment by pmac** [comment]
👍 thanks again Alex!

**Pull request merged**